docs/libraries/typespec-client-generator-core/rules/CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Contributing Linter Rules to the @azure-tools/typespec-client-generator-core Package
+
+For information about linter rules in typespec in general, view [here][generic-linter]
+
+In the `@azure-tools/typespec-client-generator-core` library, we have two main types of rules:
+
+1. Generic rule that applies to all emitted languages
+2. Specific rule that applies to a subset of all language: it can even apply to just one language
+
+The process for adding a rule starts off the same for both: for language-specific rules, there is an extra step
+
+1. Write the rule in `typespec-azure/packages/typespec-client-generator-core/src/rules/[rule-name].rule.ts
+2. Add reference to the rule in the `rules` array in [`typespec-azure/packages/typespec-client-generator-core/src/linter.ts`][tcgc-linter]
+   - This will automatically add it to a ruleset called `:all` for `@azure-tools/typespec-client-generator-core`
+3. Add the rule to the enable list for [`data-plane.ts`][data-plane-ruleset] and/or [`resource-manager.ts`][resource-manager-ruleset] in the [rulesets][rulesets] package. You can set `enable` to `false` here, if you want to delay enabling
+
+**If you are adding a language-specific rule**, you will also need this extra step 4. Add reference to the rule in the `[language]Rules` array in [`typespec-azure/packages/typespec-client-generator-core/src/linter.ts`][tcgc-linter]
+
+For Azure generations then, all rules, including all language-specific rules, will be run on the specs.
+For unbranded generations, since we've added the rules into specific `best-practices:[language]` rulesets, you can explicitly specify a subset of rules in your `tsp-config.yaml`, i.e. if I only want Python best-practices, I could add this in my `tsp-config.yaml`:
```

Comment:
This might create a bad situation where someone can turn off csharp rules but in reality they cannot.  I think we might want to consider what flexibility a user would have over tcgc rules when running in azure context.

Reply:
every azure service is enforced to have one of the 2 rules set enabled by the spec repo pipeline.

Reply:
With rulesets, we have to give users the ability to turn them off because they are ultimately warnings. After talking with @timotheeguerin, you can't have error linter rules, only warnings. If there is a case where you explicitly want to error in csharp for a csharp-specific issue, the csharp emitter itself can throw a fatal error with `reportDiagnostic`, but the fatal error should have an analog in the linter warnings in tcgc that says "if you do this in csharp and ignore me, the csharp emitter will fail"